openff/toolkit/topology/topology.py

```diff
@@ -53,13 +53,17 @@
 # =============================================================================================
 
 
-def _TOPOLOGY_DEPRECATION_WARNING(old_method, new_method):
+def _topology_deprecation(old_method, new_method):
     warnings.warn(
         f"Topology.{old_method} is deprecated. Use Topology.{new_method} instead.",
-        DeprecationWarning,
+        TopologyDeprecationWarning,
     )
 
 
+class TopologyDeprecationWarning(UserWarning):
```

Comment:
This might be too aggressive, but on the other hand, it's invisible to users if it's it a `DeprecationWarning`:

```shell
(openff-dev) [openforcefield] git diff | cat                                                  test-deprecated-api  ✭ ✱
diff --git a/openff/toolkit/topology/topology.py b/openff/toolkit/topology/topology.py
index 7f743058..535981d5 100644
--- a/openff/toolkit/topology/topology.py
+++ b/openff/toolkit/topology/topology.py
@@ -60,7 +60,7 @@ def _topology_deprecation(old_method, new_method):
     )


-class TopologyDeprecationWarning(UserWarning):
+class TopologyDeprecationWarning(DeprecationWarning):
     """Warning for deprecated portions of the Topology API."""


(openff-dev) [openforcefield] python                                                          test-deprecated-api  ✭ ✱
Python 3.8.12 | packaged by conda-forge | (default, Jan 30 2022, 23:33:09)
[Clang 11.1.0 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from openff.toolkit import Molecule
>>> topology = Molecule.from_smiles("O").to_topology()
/Users/mwt/software/openforcefield/openff/toolkit/utils/openeye_wrapper.py:1032: UnitStrippedWarning: The unit of the quantity is stripped when downcasting to ndarray.
  positions[off_atom_index, :] = off_atom_coords
>>> topology.topology_bonds
<generator object Topology.bonds at 0x7fd470170900>

Reply:
Woah, great catch. Thanks for noticing that.